src/ds++/atomics.hh

```diff
@@ -0,0 +1,71 @@
+//----------------------------------*-C++-*----------------------------------//
+/*!
+ * \file   ds++/atomics.hh
+ * \author Tim Kelley
+ * \date   Thursday, Sept. 6, 2018, 10:50 am
+ * \brief  Header file for atomic functions on floatint-point (until C++20)
+ * \note   Copyright (C) 2018 Los Alamos National Security, LLC.
+ *         All rights reserved. */
+//---------------------------------------------------------------------------//
+
+#ifndef ATOMICS_HH
+#define ATOMICS_HH
+
+#include <atomic>
+#include <type_traits> // std::is_floating_point_v
+
+namespace rtt_dsxx {
+
+/**\brief atomically add arg to a.
+ * \tparam FpT: a floating point type (integer types in std lib since C++11).
+ * \param a: an atomic of type FpT that will be updated.
+ * \param arg: quantity to add to a
+ * \param m_o: std memory order, default is memory_order_relaxed
+ * \return value of a after update
+ * \remark: By default, uses memory_order_relaxed, meaning (I think) that other
+ * atomic operations on 'a' can be moved before or after this one.
```

Comment:
Yeah, that's my reading too. I think this is what we want for our default behavior. I like that it's a default argument. If for some reason we want to get into ordering for reproducibility we could make this some kind of configurable thing like `FMA` (I sincerely hope we don't have to do that).